### PR TITLE
Scratch/add versions

### PR DESCRIPTION
**Pull request by xolian**
* refactored Project().create, upsert
* added in Version public class

### CODE REVIEW

File: fortifyapi/client.py [hunk 1]
```diff
@@ -234,53 +239,53 @@ def create(self, project_name, version_name, project_id=None, description="Creat
...
-                'name': version_name,
-                'description': description,
-                'active': active,
-                'committed': committed,
-                'project': {
-                    'id': project_id,  # if this is None it will create the project
-                    'name': project_name,
+        # Test if Project Version exists
+        if self.versions.test(project_name, version_name):
+            exit("Project: {} Version: {} exists!".format(project_name, version_name))
```
Comment (vetsin) [inline fortifyapi/client.py:244]: throw an exception, people could use this anywhere and you do not want a process exiting.

**Review by vetsin — CHANGES_REQUESTED** [review]
seems good assuming tests pass minus the exit()

File: fortifyapi/client.py (hunk 1 above)
Comment (xolian) [reply fortifyapi/client.py:244]: The self.versions.test endpoint is a bool.  Therefore, the create is either we create true or not false.  if we don't exit on an existing Project/Version the func returns a NoneType.

Comment (vetsin) [reply fortifyapi/client.py:244]: raise an exception, do not exit. Exit is a syscode and will terminate a process. You want to be able to catch it.

**Pull request merged**